src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -485,15 +342,18 @@ public boolean doesSchematicBlockEqualWorldBlock()
 
         //list of things to only check block for.
         //For the time being any flower pot is equal to each other.
-        if (schematicBlock instanceof BlockDoor || schematicBlock == Blocks.flower_pot
-              || (schematicBlock instanceof BlockStairs && BlockStairs.isSameStair(world, worldPos, schematicBlockState)))
+        if (structureBlock instanceof BlockDoor || structureBlock == Blocks.FLOWER_POT)
+        {
+            return structureBlock == worldBlockState.getBlock();
+        }
+        else if (structureBlock instanceof BlockStairs && structureBlockState == worldBlockState)
         {
-            return schematicBlock == worldBlockState.getBlock();
+            return true;
         }
 
         //had this problem in a super flat world, causes builder to sit doing nothing because placement failed
         return worldPos.getY() <= 0
-                 || schematicBlockState == worldBlockState;
+                 || structureBlockState == worldBlockState;
     }
 
     /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'worldPos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)